### PR TITLE
Improve image figure widths, remove inline styles

### LINKED ISSUE #3945: Right-align small images pinned to top-left of boxblock
GB 1.9.0

## Issue Overview
Images are pinned to the top-left of their containing blocks, even when set as right-aligned.

## Steps to Reproduce (for bugs)
1. Open demo content.
2. Resize an image to be smaller than the width of the image toolbar.
3. Right align image.
4. Toggle between selecting the image and not.

Notice the jumping image.

## Expected Behavior
If an image is right-aligned, keep the image pinned to the top right of it's container when the container needs to stay wide enough to allow for all of the tools.

## Current Behavior
Toggling selection of small right-aligned images gives a jarring jump, compared to small left-aligned images.

## Possible Solution
Pin right-aligned images to the right side of it's container.

## Screenshots / Video
gb-right-aligned-jump

## Related Issues and/or PRs
This issue is probably a more fundamental version of #3944 which reports on lack of pinning to the right of right-aligned images when resizing.

## Todos
- [ ] Tests
- [ ] Documentation

### LINKED ISSUE #5567: Make the max-width for images filterable
Hi,

I am just working on implementing theme styles for the Gutenberg editor and found the following issue: the images get a `max-width` of 608 px (the default width of the Gutenberg blocks). But I increased the max width of the blocks to the maximum content width of my theme, which is over 608 pixels.

Now I could overwrite the inline style via CSS and `!important` but I think it would be nicer to have the ability of filtering the `maxWidth` value that is set here: https://github.com/WordPress/gutenberg/blob/ee3a61c6038c67f3d787a868ad8f02a119d38faf/editor/components/provider/index.js#L51

The comment above the settings constant says, »You can override any default settings when calling initializeEditor«, but I have no idea how to call that again without getting an error.

I tried this:

```
wp.editPost.initializeEditor( 'editor', window._wpGutenbergPost, {'maxWidth': 800 );
```

But that does not work (I did not really expect it either).

Is there already a way to filter those default settings? If not, would it be possible to add a simple PHP filter that filters the setting’s array that is created in the `/lib/client-assets.php` (https://github.com/WordPress/gutenberg/blob/8a4d644a525c1463f469e525d5d31ad36f44e23c/lib/client-assets.php#L910) or something like that?

Thanks,
Florian

### PR DESCRIPTION
**Pull request by jasmussen**
As themes try to accommodate the new image figure syntax in Gutenberg, there are increasingly issues with the `figure` being a block element whereas `img` traditionally being an inline block. The bigger problem is that when we float things, and an image isn't explicitly resized, we set an inline max-width, which can only be overridden using `!important`, which you should never ever use.

GIF:

images

As such, this PR removes those inline styles. In a way, it is an almost identical copy to Gary's PR in #5209, which this is meant to replace (apologies, and you should have all the props). My concerns noted in https://github.com/WordPress/gutenberg/pull/5209#issuecomment-367936828 are not technically addressed, but in order to move on, these are perhaps best addressed separately.

Notably those issues are specific to _floats_. Specifically that if we use the float trick in https://codepen.io/joen/pen/oEYVXB to accommodate both floats and wide images in a fairly elegant way (the editor itself uses this also), we essentially need yet another wrapping element. In other words, merging in https://github.com/WordPress/gutenberg/pull/5460 would address those concerns — but we have to be okay with adding yet another wrapping element. This is also tracked in https://github.com/WordPress/gutenberg/issues/5292

This PR fixes #3945 and #5213. I think it also fixes #5567 by making that core issue moot.

CC: also @aduth and @samikeijonen, as I think you were working on a separate PR that accomplished some of the same as this. 

---

To summarize, what works well with this branch

- floats
- wide and fullwide alignments
- themes can handle floats better, and legacy themes mostly just work (though with some edgecases when captions are added)
- no need to use !important in themes

What sort of regresses in this branch is that if you insert a huge image, bigger than the main column, and float it, well it doesn't really look like it floats. You have to resize it first. This is because I removed the max-width. This is similar to how it's always worked, though perhaps not as elegant.

### CODE REVIEW

**Comment by noisysocks** [comment]
I pulled this and played with inserting image blocks and setting them to have different alignments. Everything there seems to work great, both in the editor and on the front-end!

> What sort of regresses in this branch is that if you insert a huge image, bigger than the main column, and float it, well it doesn't really look like it floats. You have to resize it first. This is because I removed the max-width. This is similar to how it's always worked, though perhaps not as elegant.

Yeah, this presents itself in a pretty nasty way when you try to align Cover Images, Galleries, Embeds blocks, etc. Some examples:

| Block | Before | After |
| --- | --- | --- |
| Cover Image | cover image before | cover image after |
| Gallery | gallery before | gallery after |
| Embed | embed before | embed after |

**Comment by jasmussen** [comment]
Awesome catch on those blocks, thank you for the review Robert. I will address those.

**Comment by jasmussen** [comment]
Okay I addressed your feedback, I believe this should be good for another review.

**Review by noisysocks — APPROVED** [review]
Nice!
